packages/dd-trace/src/debugger/devtools_client/condition.js

```diff
@@ -0,0 +1,237 @@
+function compile (node) {
+  if (node === null || typeof node === 'number' || typeof node === 'boolean' || typeof node === 'string') {
+    return JSON.stringify(node)
+  }
+
+  const [type, value] = Object.entries(node)[0]
+
+  if (type === 'not') {
+    return `!(${compile(value)})`
+  } else if (type === 'len' || type === 'count') {
+    return getSize(compile(value))
+  } else if (type === 'isEmpty') {
+    return `${getSize(compile(value))} === 0`
+  } else if (type === 'isDefined') {
+    return `(() => {
+      try {
+        ${value}
+        return true
+      } catch {
+        return false
+      }
+    })()`
+  } else if (type === 'instanceof') {
+    return `Function.prototype[Symbol.hasInstance].call(${value[1]}, ${compile(value[0])})`
+  } else if (type === 'ref') {
+    if (value === '@it') {
+      return '$dd_it'
+    } else if (value === '@key') {
+      return '$dd_key'
+    } else if (value === '@value') {
+      return '$dd_value'
+    } else {
+      return assertIdentifier(value)
+    }
+  } else if (Array.isArray(value)) {
+    const args = value.map(compile)
+    switch (type) {
+      case 'eq': return `(${args[0]}) === (${args[1]})`
+      case 'ne': return `(${args[0]}) !== (${args[1]})`
+      case 'gt': return `${guardAgainstCoercionSideEffects(args[0])} > ${guardAgainstCoercionSideEffects(args[1])}`
+      case 'ge': return `${guardAgainstCoercionSideEffects(args[0])} >= ${guardAgainstCoercionSideEffects(args[1])}`
+      case 'lt': return `${guardAgainstCoercionSideEffects(args[0])} < ${guardAgainstCoercionSideEffects(args[1])}`
+      case 'le': return `${guardAgainstCoercionSideEffects(args[0])} <= ${guardAgainstCoercionSideEffects(args[1])}`
+      case 'any': return iterateOn('some', ...args)
+      case 'all': return iterateOn('every', ...args)
+      case 'and': return `(${args.join(') && (')})`
+      case 'or': return `(${args.join(') || (')})`
+      case 'startsWith': return `String.prototype.startsWith.call(${assertString(args[0])}, ${assertString(args[1])})`
+      case 'endsWith': return `String.prototype.endsWith.call(${assertString(args[0])}, ${assertString(args[1])})`
+      case 'contains': return `((obj, elm) => {
+          if (${isString('obj')}) {
+            return String.prototype.includes.call(obj, elm)
+          } else if (Array.isArray(obj)) {
+            return Array.prototype.includes.call(obj, elm)
+          } else if (obj instanceof Object.getPrototypeOf(Int8Array)) {
+            return Object.getPrototypeOf(Int8Array.prototype).includes.call(obj, elm)
+          } else if (obj instanceof Set) {
+            return Set.prototype.has.call(obj, elm)
+          } else if (obj instanceof WeakSet) {
+            return WeakSet.prototype.has.call(obj, elm)
+          } else if (obj instanceof Map) {
+            return Map.prototype.has.call(obj, elm)
+          } else if (obj instanceof WeakMap) {
```

Comment:
Instead of using instanceof, should we use `util.types.isWeakMap()` instead? Those work across realms, instanceof can't handle things across realms.

Reply:
Hmm... we don't have access to `require` in the scope that this code runs, so we can't easily load `util`. I made an exception for `isProxy` and added it to `process` so that I could access it here, because there's no other way to check if something is a proxy.

1. So we can either decide we don't care about realm support
2. Or we can add the specific methods we need to `process` just like I did with `isProxy`
3. Or we can just go all in and add all of `util.types` there

Are there any cons to just doing no 3 that you can think of?

Reply:
Based on IRL conversation, I've added `util.types` to the `process` symbol so we have access to all these and I've updated the code accordingly 👍